lib/scoring.rb

```diff
@@ -1,9 +1,89 @@
 module Scrabble
   class Scoring
+
     def self.score(word)
+
+      # Handling "terminal if's" that return nil
+      if word == nil || word == ""  || word =~ /[^a-zA-Z]/
+        return nil
+      elsif word.length > 7
+        return nil
+      end
+
+      # Take input as a string, split the word, downcase it, store word in an array called "letter_array"
+      letter_array = []
+      letter_array = word.downcase.split(//)
+      # SCORECARD AKA 'provided_words_values'
+      value_1_array = ["a", "e", "i", "o", "u", "l", "n", "r", "s", "t"]
+      value_2_array = ["d", "g"]
+      value_3_array = ["b", "c", "m", "p"]
+      value_4_array = ["f", "h", "v", "w", "y"]
+      value_5_array = ["k"]
+      value_8_array = ["j", "x"]
+      value_10_array = ["q", "z"]
+
+      # Gives 50 extra points to a valid word with 7 letters
+      if letter_array.length == 7
+        word_score = 50
+      else
+        word_score = 0
+      end
+
+      letter_array.each do |letter|
+        if value_1_array.include?(letter)
+          word_score += 1
+        elsif value_2_array.include?(letter)
+          word_score += 2
+        elsif value_3_array.include?(letter)
+          word_score += 3
+        elsif value_4_array.include?(letter)
+          word_score += 4
+        elsif value_5_array.include?(letter)
+          word_score += 5
+        elsif value_8_array.include?(letter)
+          word_score += 8
+        elsif value_10_array.include?(letter)
+          word_score += 10
+        end
+      end
+
+      return word_score
     end
 
     def self.highest_score_from(array_of_words)
-    end
-  end
-end
+      # Handle empty input
+      if array_of_words.length == 0
+        return nil
+      end
+
+      scored_words_hash = {}
+      winner_hash = {}
+
+      array_of_words.each do |provided_word|
+        score = Scrabble::Scoring.score(provided_word)
+        scored_words_hash[provided_word] = score
+      end
+
+      # Highest score values
+      highest_score_array = scored_words_hash.max_by{|provided_word,score| score}
+      puts "This is the highest score value: #{highest_score_array[1]}"
+      highest_score_value = highest_score_array[1]
```

Comment:
This method works, but I think I would argue you're over-using enumerables here, and it makes the code more complex than it needs to be. My strategy, especially when I see that the enumerable methods aren't quite matching up to the problem I'm solving, is to write (or pseudocode) it as simply as I can using an `each` loop, and then try and spot patterns where an enumerable might make things cleaner.

Refer to the in-class implementation we did as an example of this.